bin/picom-trans

```diff
@@ -46,194 +55,311 @@ print_usage() {
+      (d) action=delete;;
+      (t) action=toggle;;
+      (r) action=reset;;
+      (g) action=get;;
+      (s) winidtype=;         winid=;;
+      (n) winidtype=-name;    winid="$OPTARG";;
+      (w) winidtype=-id;      winid="$OPTARG";;
+      (c) winidtype=-id;      winid="$(get_focused_window_id)";;
+      (\?) exit 1
+    esac
   done
-  exit 0
-fi
+} #}}
 
-# Get ID of the target window
-if test -z "$wprefix"; then
-  treeout=$(xwininfo -children -frame)
-else
-  test "$wprefix" = '-id' \
-    && ! echo "$window" | grep -Eiq '^[[:space:]]*(0x[[:xdigit:]]+|[[:digit:]]+)[[:space:]]*$' \
-    && echo 'Bad window ID.' && exit 1
-  treeout=$(xwininfo -children $wprefix "$window")
-fi
 
-wid=$(echo "$treeout" | sed -n 's/^xwininfo:.*: \(0x[[:xdigit:]]*\).*$/\1/p')
+# NOTE: The code inside the following function has been taken directly from the
+# old version of picom-trans.  Tread cautiously!
+get_target_window_id()
+{ #{{
 
-if test -z "$wid"; then
-  echo 'Failed to find window.'
-  exit 1
-fi
+  # Get ID of the target window
+  if test -z "$winidtype"; then
+    treeout=$(xwininfo -children -frame)
+  else
+    test "$winidtype" = "-id" \
+      && ! echo "$winid" | grep -Eiq '^[[:space:]]*(0x[[:xdigit:]]+|[[:digit:]]+)[[:space:]]*$' \
+      && echo "Bad window ID." >&2 \
+      && exit 1
+    treeout=$(xwininfo -children $winidtype "$winid")
+  fi
 
-# Make sure it's not root window
-if echo "$treeout" | fgrep -q 'Parent window id: 0x0'; then
-  echo 'Cannot set opacity on root window.'
-  exit 1
-fi
+  wid=$(echo "$treeout" | sed -n 's/^xwininfo:.*: \(0x[[:xdigit:]]*\).*$/\1/p')
 
-# If it's already the topmost window
-if echo "$treeout" | grep -q 'Parent window id: 0x[[:xdigit:]]* (the root window)'; then
-  topmost=$wid
-else
-  # Get the whole window tree
-  treeout=$(xwininfo -root -tree)
+  if test -z "$wid"; then
+    echo "Failed to find window." >&2
+    exit 1
+  fi
 
-  if test -z "$treeout"; then
-    echo 'Failed to get root window tree.'
+  # Make sure it's not root window
+  if echo "$treeout" | grep -Fq "Parent window id: 0x0"; then
+    echo "Cannot set opacity on root window." >&2
     exit 1
   fi
 
-  # Find the line number of the target window in the window tree
-  lineno=$(echo -n "$treeout" | grep -nw "^\s*$wid" | head -n1 | cut -d ':' -f 1)
+  # If it's already the topmost window
+  if echo "$treeout" | grep -q 'Parent window id: 0x[[:xdigit:]]* (the root window)'; then
+    topmost=$wid
+  else
+    # Get the whole window tree
+    treeout=$(xwininfo -root -tree)
+
+    if test -z "$treeout"; then
+      echo "Failed to get root window tree." >&2
+      exit 1
+    fi
+
+    # Find the line number of the target window in the window tree
+    lineno=$(printf "%s" "$treeout" | grep -nw "^\s*$wid" | head -n1 | cut -d ':' -f 1)
 
-  if test -z "$lineno"; then
-    echo 'Failed to find window in window tree.'
+    if test -z "$lineno"; then
+      echo 'Failed to find window in window tree.' >&2
```

Comment:
Probably missed this line when you were changing the quotes to `"`.

Reply:
4a8dde0